doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+Security Kernel Loader
+######################
+
+The Security Kernel Loader (SKL) is an implementation of a Secure Loader for
+the SKINIT CPU instruction. The SKL loader takes a minimal approach relying on
+the DRTM Preamble, e.g. the system bootloader, to load into memory and
+configure the Security Kernel that the SKL will measure and execute.
+
+Memory Layout
+*************
+
+The DRTM Preamble is responsible for allocating a contiguous block of physical
+memory and will ensure it is marked as reserved in the memory map. The
+necessary size for the allocation will be found in the SKL header. The DRTM
+Preamble will place the SKL at the beginning of the allocation.
+
+Below is the layout of how the memory allocation is segmented::
+
+      +---     Security Kernel Loader    ---+
+      |                                     |
+      v                                     v
+      +--------+----------+-------------+-------------+--------------+
+      | Header |   Code   |  Boot Tags  | IOMMU Table |  Log Buffer  |
+      +--------+----------+-------------+-------------+--------------+
+      ^                                 ^                            ^
+      |                                 |                            |
+      +                                 +                            +
+  sklbuf_base                          60kb                   header.skl_size
+
+
+SKL Header
+==========
+
+The SKL header is the AMD Secure Loader Block (SLB) header extended to meet the
+needs of the SKL. The first two fields are mandated by the SLB and the SKL adds
+two additional fields which can be seen in the following representative struct,
+::
+
+    typedef struct __packed sl_header {
+        u16 entry_point;        /* Code Entry */
+        u16 boot_tags_offset;   /* RoT Measurement Length */
+        u16 info_table_offset;  /* Offset for the SKL Info Table */
+        u16 alloc_size;         /* Allocation Size */
+    } sl_header_t;
+
+SKL Allocation Size
+-------------------
+
+The length of Boot Tags are a runtime-determined size but cannot exceed the
+last possible start address for the IOMMU Table, that being 61440 byte offset
+from the beginning of the allocation. For build time calculation of the SKL
+allocation size, the IOMMU table will be considered to start at offset 61440.
+This renders the build time calculation to,
+
+    skl_size = 61440 + PAGETABLE_SIZE + LOGBUFFER_SIZE
+
+Where the default for PAGETABLE_SIZE is 28kb and LOGBUFFER_SIZE is 4kb.
```

Comment:
Which pagetable is it, CPU one? If yes, then this is wrong, we can't put it outside of SLB.